src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts

```diff
@@ -157,6 +159,7 @@ export class PoolListComponent extends ListWithDetails implements OnInit {
       {
         prop: 'crush_rule',
         name: $localize`Crush Ruleset`,
+        headerTemplate: this.headerTooltipTpl,
```

Comment:
> With the current changes to add a tooltip to 'Crush Ruleset' column header (to show description of Crush Ruleset), it loses the default properties of a table column (sortable column, bold text, etc), as I have used headerTemplate to add the tooltip. So how can I satisfy the requirement (of adding tooltip to header), without changing the default properties of the column, any suggestion?
> 
> Current view:
> ![Screenshot from 2020-10-13 15-54-16](https://user-images.githubusercontent.com/23611106/95848954-63fbaf80-0d6c-11eb-9974-49b7728e9258.png)

@avanthakkar if the only purpose of this template is for adding a tooltip to the column header, and that might be a desirable option for other columns, why not extending `CdTableColumn` to support this?

Reply:
headerTemplate is already a member of TableColumn ,similar to name,prop, cellTemplate, etc

Reply:
And it basically breaks the rest of the functionality, right? So why using a template and redefine the column header from scratch when you just want to display a tooltip in a regular column header?

Reply:
Yes, this approach will not work. But, what other approach can be used to access a column header and then add any functionality to it?

Reply:
> Yes, this approach will not work. But, what other approach can be used to access a column header and then add any functionality to it?

Any suggestions on this @tspmelo ?

Reply:
Found this discussion https://github.com/swimlane/ngx-datatable/issues/808 where they talk a bit about this, but its not exactly our situation. Maybe it would be possible to adapt some of the suggestions to get it working.

But the best solution would be to implement this feature directly in `ngx-datatable` by implementing this issue: https://github.com/swimlane/ngx-datatable/issues/1347
(I might have a look at it later today to check how difficult this is)

Reply:
I will take a look at it. Thanks!

Reply:
I think following this approach similar as https://github.com/swimlane/ngx-datatable/issues/808 (assuming will work in our situation), if we want to apply it to several other or maybe all column header to add it's description in tooltip, we need to do repeat the similar piece of code for every column, right?

Reply:
@avanthakkar by having a look at [ngx-datatable's header-cell.component.ts](https://github.com/swimlane/ngx-datatable/blob/e1fdf0bc74091a76f221418d96b7e8f2911844b3/projects/swimlane/ngx-datatable/src/lib/components/header/header-cell.component.ts#L31), it looks like the column name allows for HTML, so could you just try an inject a tooptip component (probably that will require a [component factory](https://stackoverflow.com/questions/57505273/how-to-render-component-by-adding-the-component-using-innerhtml)) or try with some HTML/CSS only tooltip?

And... why not "simply" opening a PR in ngx-datatable project with a new tooltip input on the column header cell? However, in that case, I'd go with this PR without tooltips...

Reply:
@avanthakkar @tspmelo would it be possible that if a `tooltip: "this is that"` property is added to the column definition, our datatable helper would add as column name `<span ngbTooltip="this is that">Column Header</span>` so it ends up injected in the `innerHTML`? Any clue if Angular would process the `ngbTooltip` directive?

Reply:
This works for the header name, but the tooltip fails. It shows the full string of html.

I think the best solution would be to provide the custom headerTemplate, but instead of just adding a tooltip to the name, we should add an info icon, that would displays the tooltip with all info.

Reply:
Yes, that's a good idea of adding a info icon. I thought the same initially but dropped the idea as we already have sort icon and adding one more icon, does it look good  form UX perspective?

Reply:
I had the same problem on one of my PR's and haven't found a good solution for it.

Does the suggestion with the icon work? - If not, I'm wondering if it's a workaround to add the information to the pool creation dialogue, as a tooltip is very limited in my opinion.

Reply:
Thanks @s0nea for the feedback! I will try first with the info icon and if it works and everyone agree with the design of it, then will follow the approach. If not, I think the workaround you suggested LGTM!